src/DataStructures/Tensor/EagerMath/Norms.hpp

```diff
@@ -0,0 +1,213 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <string>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Utilities/Functional.hpp"
+#include "Utilities/Numeric.hpp"
+#include "Utilities/Requires.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TypeTraits.hpp"
+
+namespace L2Norm_detail {
+template <typename DataType, typename Symm, typename IndexList>
+Scalar<DataType> pointwise_l2_norm_square(
+    const Tensor<DataType, Symm, IndexList>& tensor) noexcept {
+  auto pointwise_l2_normsq = make_with_value<Scalar<DataType>>(tensor, 0.);
+  for (auto tensor_element = tensor.begin(); tensor_element != tensor.end();
+       ++tensor_element) {
+    // In order to handle tensory symmetries, we multiply the square of each
+    // stored component with its multiplicity
+    get(pointwise_l2_normsq) +=
+        tensor.multiplicity(tensor_element) * square(*tensor_element);
+  }
+  return pointwise_l2_normsq;
+}
+}  // namespace L2Norm_detail
+
+// @{
+/*!
+ * \ingroup TensorGroup
+ * \brief Compute point-wise Frobenius norm of arbitrary Tensors.
+ *
+ * \details
+ * At each point \f$p\f$ in the element, this function computes the
+ * point-wise Frobenius norm(s) of a given Tensor with arbitrary rank (or a
+ * Variables of Tensors). If the Tensor \f$A\f$ has rank \f$n\f$ and
+ * dimensionality \f$D\f$, then its Frobenius norm at point \f$ p\f$ is
+ * computed as:
+ *
+ * \f{equation}
+ * ||A||_2^p =
```

Comment:
I has superscripted (instead of subscripting) `p` to disambiguate it from tensorial indices. I've now moved `p` from superscript to argument notation `(p)`, which seems right in the context of these equations.